**HW2 Titanic Data Analysis**
1. How many variables and observations? Which are Numerical/Categorical?
2. Do any of the columns have NaNs in them? What do NaNs mean?
3. How many passengers survived?
4. Is survival correlated with Fare?
5. How many passengers were alone vs. traveling with family?
6. Were people traveling alone more or less likely to survive?
7. Do the basic statistics change if you group by class?

In [19]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [20]:
file = "/Users/admin/.cache/kagglehub/datasets/yasserh/titanic-dataset/versions/1/Titanic-Dataset.csv"
df = pd.read_csv(file)

In [21]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

**1. How many variables and observations? Which are Numerical/Categorical?**


In [22]:
df.shape
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


So this dataset contains a lot of information about the titanic customers. There are 891 observations(rows) and 12 variables(columns). Some numerical values include, PassengerId, Survived, Pclass, Age, SibSp, Parch, Fare, and some categorical variables include, Name, Sex, Ticket, Cabin, Embarked.

**2. Do any of the columns have NaNs in them? What do NaNs mean?**


In [23]:
df.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In this section, there are a few columns with a few NaN's. Age has about 177 missing values, Cabin has 687 missing values, and Embarked has 2 values. NaN means not a number which represents missing or unknown data. This in the context of age, is if the passenger's age was not recorded it would be recorded as NaN.

**3. How many passengers survived?**


In [24]:
df['Survived'].value_counts()

Survived
0    549
1    342
Name: count, dtype: int64

This data set shows us that 342 passengers survived and 549 passengers did not survive. The 0 represents the people that died and 1 represents the people that survived.

**4. Is survival correlated with Fare?**


In [25]:
df.groupby('Survived')['Fare'].mean()
df[['Fare', 'Survived']].corr()

Fare  Survived
Fare      1.000000  0.257307
Survived  0.257307  1.000000

Survivors paid about $48 for their ticket, while non-survivors paid about 22 dollars. The correlation between fare and survival is about 0.26, which is a small but positive relationship. This suggests that people who paid higher fares, often those in higher classes, were more likely to survive the Titanic disaster.


**5. How many passengers were alone vs. traveling with family?**


In [26]:
df['FamilySize'] = df['SibSp'] + df['Parch']
df['Alone'] = df['FamilySize'] == 0

df['Alone'].value_counts()

Alone
True     537
False    354
Name: count, dtype: int64

About 537 passengers traveled alone, while 354 passengers traveled with their families.

**6. Were people traveling alone more or less likely to survive?**


In [27]:
df.groupby('Alone')['Survived'].mean()

Alone
False    0.505650
True     0.303538
Name: Survived, dtype: float64

Passengers traveling alone survived at a lower rate at around 30%, while those traveling with family survived at a higher rate of around 50%. This meant that traveling with a family increased their chances of survival. 

**7. Do the basic statistics change if you group by class?**

In [28]:
df.groupby('Pclass')['Survived'].mean()
df.groupby(['Pclass','Sex'])['Survived'].mean()

Pclass  Sex   
1       female    0.968085
        male      0.368852
2       female    0.921053
        male      0.157407
3       female    0.500000
        male      0.135447
Name: Survived, dtype: float64

1st class passengers had the highest survival rates, while 3rd class passengers had the lowest. Looking more closely at both class and sex together, we see that women in 1st and 2nd class had extremely high survival rates, while women in 3rd class survived about half the time. Men, on the other hand, had much lower survival chances across all classes, with 1st class men surviving less than men in 2nd and 3rd class. This shows that both class and gender played a major role in survival on the Titanic.


**8. Survival by Sex**

In [29]:
df.groupby('Sex')['Survived'].mean()

Sex
female    0.742038
male      0.188908
Name: Survived, dtype: float64

Here we see that there was a much higher survival rate for women compared to men, this makes sense as well all know the women and children first rule, and this was a strong predictor on whether or not someone survived.

**9.Survival by Age Group**

In [30]:
df['AgeGroup'] = pd.cut(df['Age'], bins=[0,12,18,60,80], labels=['Child','Teen','Adult','Senior'])
df.groupby('AgeGroup', observed=False)['Survived'].mean()

AgeGroup
Child     0.579710
Teen      0.428571
Adult     0.388788
Senior    0.227273
Name: Survived, dtype: float64

When we group passengers by age, we see that children had the highest chance to survive, followed by teens and adults, then finally seniors. This shows that younger people had priority during evacuation, and they also naturally had a better chance to survive.

**Final Conclusion**

Overall, only about 38% of the passengers survived. Women and Children had the highest survival rates while men and seniors had the lowest. Wealth and class also mattered, as those in 1st class had a higher survival rate than 3rd class. Passengers travelling with family also had a higher chance of surviving. In summary, survival on the Titanic was strongly effected by gender, age, family ties, and social class. This shows how social structures can impact the effect of a disaster. 